src/input.h

```diff
@@ -13,6 +13,7 @@
 #include <algorithm>
 #endif
 
+#include "action.h"
```

Comment:
```suggestion
enum action_id : int;
```
A forward declaration of the enum is enough for its usage in the function declaration. (The header is still needed within the cpp file.)

Reply:
Thank you, @BevapDin. I will make the change.